src/Components/Components/src/RenderTree/RenderTreeFrame.cs

```diff
@@ -216,19 +216,19 @@ private RenderTreeFrame(int sequence, int elementSubtreeLength, string elementNa
         }
 
         // Component constructor
-        private RenderTreeFrame(int sequence, int componentSubtreeLength, Type componentType, ComponentState componentState, object componentKey)
+        private RenderTreeFrame(int sequence, int componentSubtreeLength, Type componentType, int componentId, ComponentState componentState, object componentKey)
             : this()
         {
             Sequence = sequence;
             FrameType = RenderTreeFrameType.Component;
             ComponentSubtreeLength = componentSubtreeLength;
             ComponentType = componentType;
             ComponentKey = componentKey;
+            ComponentId = componentId;
 
             if (componentState != null)
             {
                 ComponentState = componentState;
-                ComponentId = componentState.ComponentId;
```

Comment:
Also FYI @SteveSandersonMS

Reply:
I'd like to understand what the change is required for. Before this change, it was impossible to have a nonempty `ComponentId` without also having a `ComponentState`, but this change would allow that situation to occur. Why do we want that to be possible?

None of the static factory methods that call this constructor make use of this possibility. Am I missing the reason why this extra param is required?

Reply:
We'll have to ask @rynowak his thoughts on this one. I've undone it locally for now and all seems to work.